### ISSUE TITLE
FNX-12352 ⁃ [Bug]Native code crash when quickly opening the Top sites

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#11419]
## Steps to reproduce
1. Quickly open one by one the default top sites

### Expected behavior
The websites are properly opened 

### Actual behavior
Native code crash

### Device information
* Android device: OnePlus A3 (Android 6.0.1)
* Fenix version: Nightly Build 20610 from 6/10

### Notes
► Video
![20200610-124548](https://user-images.githubusercontent.com/51314259/84256321-6c24b380-ab1c-11ea-80b7-383caec65ea8.gif)

d4ae4e98-6728-4d26-a1d9-c50f6cf3ff21

## <native crash>
<native crash>

►Log 

```kotlin
06-10 13:10:27.299 28877-28890/? E/mozac/CrashReporter: CrashHandlerService received native code crash
06-10 13:10:27.301 1302-1707/? D/ConnectivityService: releasing NetworkRequest NetworkRequest [ id=170, legacyType=-1, [ Transports: WIFI Capabilities: NOT_RESTRICTED&TRUSTED&NOT_VPN] ]
06-10 13:10:27.302 28877-28890/? I/mozac/CrashReporter: Received crash: NativeCodeCrash(minidumpPath=/data/user/0/org.mozilla.fenix.nightly/files/mozilla/84e5x2cm.default/minidumps/0ddb031c-13a3-c42b-3a19-086c885be51b.dmp, minidumpSuccess=false, extrasPath=/data/user/0/org.mozilla.fenix.nightly/files/mozilla/84e5x2cm.default/minidumps/0ddb031c-13a3-c42b-3a19-086c885be51b.extra, isFatal=true, breadcrumbs=[], uuid=51d7defb-aa8a-423e-b48f-678ad6d80da3)
06-10 13:10:27.339 28877-28890/? I/mozac/CrashReporter: Showing prompt
06-10 13:10:27.341 1302-1779/? I/ActivityManager: START u0 {flg=0x10018000 cmp=org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras)} from uid 10277 pid 28877 on display 0
06-10 13:10:27.350 28517-28548/? I/mozac/CrashReporter: Crash report submitted to 1 telemetry services
06-10 13:10:27.376 28877-28890/? I/art: System.exit called, status: 0
06-10 13:10:27.376 28877-28890/? I/AndroidRuntime: VM exiting with result code 0, cleanup skipped.
06-10 13:10:27.427 1302-3130/? I/ActivityManager: Process org.mozilla.fenix.nightly:mozilla.components.lib.crash.CrashHandler (pid 28877) has died
06-10 13:10:27.427 1302-3130/? W/ActivityManager: Scheduling restart of crashed service org.mozilla.fenix.nightly/mozilla.components.lib.crash.handler.CrashHandlerService in 1000ms
06-10 13:10:27.427 1302-1342/? V/ActivityManager: killProcessGroupAsync took 0 ms for PID 28877 on thread 14
06-10 13:10:27.447 1302-2643/? D/GraphicsStats: Buffer count: 6
06-10 13:10:27.447 1302-2780/? I/WindowState: WIN DEATH: Window{63f49a6 u0 org.mozilla.fenix.nightly/org.mozilla.fenix.nightly.App}
06-10 13:10:27.447 1302-2643/? I/ActivityManager: Process org.mozilla.fenix.nightly (pid 28517) has died
06-10 13:10:27.449 28581-28581/? I/ServiceChildProcess: Service has been unbound. Stopping.
06-10 13:10:27.461 1302-2643/? V/BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@e5f6f75
06-10 13:10:27.461 1302-2643/? I/ActivityManager: Start proc 28893:org.mozilla.fenix.nightly:mozilla.components.lib.crash.CrashReporter/u0a277 for activity org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity
06-10 13:10:27.463 28581-28581/? I/Process: Sending signal. PID: 28581 SIG: 9
06-10 13:10:27.537 1302-1326/? I/ActivityManager: Process org.mozilla.fenix.nightly:tab0 (pid 28581) has died
06-10 13:10:27.587 1302-2189/? W/InputMethodManagerService: Got RemoteException sending setActive(false) notification to pid 28517 uid 10277
06-10 13:10:27.640 1302-1342/? V/ActivityManager: killProcessGroupAsync took 192 ms for PID 28517 on thread 14
06-10 13:10:27.641 1302-1342/? V/ActivityManager: killProcessGroupAsync took 1 ms for PID 28581 on thread 14
06-10 13:10:27.641 720-720/? I/Zygote: Process 28517 exited cleanly (11)
06-10 13:10:27.759 28893-28893/? D/AppTracker: App Event: start
06-10 13:10:27.770 28893-28909/? D/OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
06-10 13:10:27.815 28893-28909/? I/Adreno: QUALCOMM build                   : 19f8f74, Iac38d51750
    Build Date                       : 06/06/16
    OpenGL ES Shader Compiler Version: XE031.07.00.01
    Local Branch                     : case2489916_Unmap_crash
    Remote Branch                    : 
    Remote Branch                    : 
    Reconstruct Branch               : 
06-10 13:10:27.820 28893-28909/? I/OpenGLRenderer: Initialized EGL, version 1.4
06-10 13:10:27.848 28893-28909/? E/HAL: load: gralloc.msm8996.so
06-10 13:10:27.939 28893-28893/? D/AbstractTracker: Event success
06-10 13:10:27.954 1302-1365/? I/ActivityManager: Displayed org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity: +503ms
06-10 13:10:30.081 2076-2246/? I/VoiceInputManagerWrappe: VoiceInputManagerWrapper.shutdownVoiceInternal():95 shutdownVoiceInternal()

```

**Comment by kbrosnan** [comment]
@AndiAJ please get a crash stats ID from the about Firefox cashes page. Details at https://github.com/mozilla-mobile/fenix/wiki/Logging-Crash-Information#retrieving-crash-reports-from-the-application

**Comment by AndiAJ** [comment]
Hey @kbrosnan , I've managed to **repro** the issue on Nightly 200616 from 6/16 using a OnePlus A3 (Android 6.0.1)

❌ There's no Socorro link in the About Firefox → Crashes 
❓ Not sure what **da38d411-75fd-4f30-96f2-d6ac0968ade3** stands for  

![rsz_screenshot_20200621-014009](https://user-images.githubusercontent.com/51314259/84863180-397b3d80-b07d-11ea-9409-befe7ab2081c.png)

Please find attached the [Logcat](https://github.com/mozilla-mobile/fenix/files/4790700/nativaC.txt)

► Video
![20200621-013927](https://user-images.githubusercontent.com/51314259/84864391-4731c280-b07f-11ea-96f9-719f2944c64a.gif)

Please review and share your thoughts.

**Comment by kbrosnan** [comment]
That looks like the behavior we have been seeing related to this crash.

**Comment by kbrosnan** [comment]
@AndiAJ  please retest on Monday or later. A fix was checked in today for BZ-1644561.

**Comment by kbrosnan** [comment]
Duplicate #11653
